src/Features/Core/Portable/ExternalAccess/UnitTesting/API/UnitTestingStackTraceService.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.StackTraceExplorer;
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.UnitTesting.API
+{
+    internal sealed class UnitTestingStackTraceService
+    {
+        private readonly Workspace _workspace;
+        private readonly IStackTraceExplorerService _stackTraceService;
+
+        public UnitTestingStackTraceService(Workspace workspace)
+        {
+            _workspace = workspace;
+            _stackTraceService = workspace.Services.GetRequiredService<IStackTraceExplorerService>();
+        }
+
+        public async Task<ImmutableArray<Frame>> TryParseAsync(string input, CancellationToken cancellationToken)
+        {
+            var result = await StackTraceAnalyzer.AnalyzeAsync(input, cancellationToken).ConfigureAwait(false);
+            return result.ParsedFrames.SelectAsArray(p => new Frame(p, _stackTraceService, _workspace));
+        }
+
+        public readonly struct Frame
+        {
+            private readonly ParsedFrame _parsedFrame;
+            private readonly IStackTraceExplorerService _stackTraceService;
+            private readonly Workspace _workspace;
+
+            public Frame(ParsedFrame parsedFrame, IStackTraceExplorerService service, Workspace workspace)
+            {
+                _parsedFrame = parsedFrame;
+                _stackTraceService = service;
+                _workspace = workspace;
+            }
+
+            public (Document? document, int lineNumber) TryGetDocumentAndLine()
+                => _stackTraceService.GetDocumentAndLine(_workspace.CurrentSolution, _parsedFrame);
```

Comment:
This should be moved to `IUnitTestingStackTraceServiceAccessor`, and be renamed to match the name of the method in `IStackTraceService`. Over time it's possible for the names to diverge (this is why the external access layer exists), but at the point of creation the layer should be as consistent as possible.